homework 8

gemansipator opened the pull request.

gemansipator committed:
Создать класс Animal и расширяющие его классы Dog, Cat, Horse.
 Класс Animal содержит переменные food, location и методы makeNoise, eat, sleep. Метод makeNoise, например, может
 выводить на консоль "Такое-то животное спит".
 Dog, Cat, Horse переопределяют методы makeNoise, eat.
 Добавьте переменные в классы Dog, Cat, Horse, характеризующие только этих животных.
 Создайте класс Ветеринар, в котором определите метод void treatAnimal(Animal animal). Пусть этот метод распечатывает
 food и location пришедшего на прием животного.
 В методе main создайте массив типа Animal, в который запишите животных всех имеющихся у вас типов. В цикле отправляйте
 их на прием к ветеринару.

gemansipator committed:
лишнее

gemansipator committed:
Task02 оздать класс Товар, имеющий переменные имя, цена, рейтинг.
        Создать класс Категория, имеющий переменные имя и массив товаров. Создать несколько объектов класса Категория.
        Создать класс Basket, содержащий массив купленных товаров.
        Создать класс User, содержащий логин, пароль и объект класса Basket. Создать объект класса User.

gemansipator committed:
Task02 поменял логику

gemansipator committed:
Task02 README.md

gemansipator committed:
Task03  Создать класс Person, который содержит:
         переменные fullName, age;
         методы move() и talk(), в которых просто вывести на консоль сообщение -"Такой-то  Person говорит".
         Добавьте два конструктора  - Person() и Person(fullName, age).
         Создайте два объекта этого класса. Один объект инициализируется конструктором Person(),
         другой - Person(fullName, age).

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task01/Animal.java:
```diff
@@ -0,0 +1,12 @@
+
+public abstract class Animal {
+    String food;
+    String location;
```
здесь нужно поставить модификатор доступа protected, чтобы наследники могли получать доступ к этим переменным. А сейчас получается, что если наследник в другой директории будет, то не получит доступ к ним.

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task01/Cat.java:
```diff
@@ -0,0 +1,31 @@
+
+public class Cat extends Animal{
+
+    int purringVolume; //Громкость мурлыкания в децибелах. Средняя 25
```
private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task01/Dog.java:
```diff
@@ -0,0 +1,34 @@
+
+public class Dog extends Animal{
+
+    int loyaltyLevel;    //Уровень преданности хозяину. от 1 до 10
```
private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task01/Horse.java:
```diff
@@ -0,0 +1,32 @@
+package site.javadev.lesson_09.task01;
+
+public class Horse extends Animal {
+    int maneLength; // длина гривы в сантиметрах
```
private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task01/Vet.java:
```diff
@@ -0,0 +1,11 @@
+    public void treatAnimal(Animal animal) {
+        System.out.println("Ветеринар осматривает " + animal);
+        System.out.println("Еда: " + animal.food);
+        System.out.println("Местоположение: " + animal.location);
```
доступ к переменным должен осуществляться через геттеры

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task02/Main.java:
```diff
@@ -0,0 +1,30 @@
+
+
+        //пустая корзина
+        Basket basket = new Basket();
```
корзина не должна быть пустой

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task02/User.java:
```diff
@@ -0,0 +1,14 @@
+public class User {
+    private String login;
+    private String password;
+    Basket basket;
```
тоже private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task02/Category.java:
```diff
@@ -0,0 +1,10 @@
+package site.javadev.lesson_09.task02;
+
+public class Category {
+    String name; // имя категории
```
private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task02/Product.java:
```diff
@@ -0,0 +1,13 @@
+public class Product {
+    String name; // имя товара
+    double price; // цена товара
+    double rating; // рейтинг товара
```
private

RinatBeybutov commented on src/main/java/site/javadev/lesson_09/task03/Person.java:
```diff
@@ -0,0 +1,22 @@
+
+public class Person {
+
+    String fullName; // полное имя человека
```
private

gemansipator committed:
Task01 protected
public abstract class Animal {
           protected String food;
           protected String location;

gemansipator committed:
Task01 private
public class Cat extends Animal{

    private int purringVolume; //Громкость мурлыкания в децибелах. Средняя 25

    public Cat(int purringVolume, String food, String location) {
        this.food = food;
        this.location = location;
        this.purringVolume = purringVolume;
    }

gemansipator committed:
Task01 private
public class Dog extends Animal{

    private int loyaltyLevel;    //Уровень преданности хозяину. от 1 до 10

    public Dog( int loyaltyLevel, String food, String location) {
        this.food = food;
        this.location = location;
        this.loyaltyLevel = loyaltyLevel;

    }

gemansipator committed:
Task01 private
public class Horse extends Animal {
    private int maneLength; // длина гривы в сантиметрах

    public Horse(int maneLength, String food, String location) {
        this.food = food;
        this.maneLength = maneLength;
        this.location = location;
    }

gemansipator committed:
Task01 доступ к переменным food и location в классе Vet осуществляется через геттеры

gemansipator committed:
Task01 корзина уже не пустая, так как она инициализируется товарами из категории iPhone.

gemansipator committed:
Task02 private
public class User {
    private String login;
    private String password;
    private Basket basket;

    public User(String login, String password, Basket basket) {
        this.login = login;
        this.password = password;
        this.basket = basket;
    }
}

gemansipator committed:
Task02 private

gemansipator committed:
Task03 private
